### PR TITLE
Cities not loading bug fixed

### PR DESCRIPTION
**Pull request by Shruti-Apte**
### **Notion :**
https://www.notion.so/saeloun/The-cities-list-loading-only-when-the-user-starts-typing-ccd736741ce741fbb3a0b6682694d7e0?pvs=4

### **What :**
- Cities not loading bug fixed
### **Why :**
- When we click on city dropdown the city list should be displayed
### **Preview :**
Before:
Screenshot 2023-05-31 at 3 18 58 PM

After:
Screenshot 2023-05-31 at 3 18 05 PM

### **Todos** (for testing):
Tested if cities are loading on selecting the city field, or changing the state field.

### CODE REVIEW

**Comment by apoorv1316** [comment]
@Shruti-Apte The same fix needs to be done on the team details page, while creating a new company and editing it.

**Comment by apoorv1316** [comment]
> @Shruti-Apte The same fix needs to be done on the team details page, while creating a new company and editing it.

@Shruti-Apte Gentle reminder

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@Shruti-Apte If a city is already selected then its not showing cities list. We have to change state or country first to change the city.
Screenshot 2023-06-05 at 5 34 21 PM

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte If a city is already selected then its not showing cities list. We have to change state or country first to change the city. Screenshot 2023-06-05 at 5 34 21 PM

There are no cities present for central district state, If you select different country and state it's working properly.

**Comment by apoorv1316** [comment]
@Shruti-Apte Can't access edit org. page. Please check all the pages once where we can create/edit address.
Screenshot 2023-06-08 at 1 17 20 PM

**Comment by apoorv1316** [comment]
@Shruti-Apte Have you updated the seed data and tested the timezone and fiscal year that we discussed yesterday?

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte Have you updated the seed data and tested the timezone and fiscal year that we discussed yesterday?

Yes, In seed data fiscal year is stored as` jan-dec`  but in code we are accepting only single month value like `Dec`,  `Nov` we need to update it from seed data

**Comment by Shruti-Apte** [comment]
@prasanthchaduvula Please review this PR.
The bug:
Earlier  the cities list on the “Setup org”, “Edit Org” & “Edit Personal Details” were loading only when the user starts typing on the field.  Also the city value was not updating if user changes the state or country.  

Bug Fix:
Now the list of cities  is populating on click of city dropdown. also if user changes the country or state the City value will be removed.

 we can connect for demo.

**Comment by prasanthchaduvula** [comment]
@Shruti-Apte Please update the seed data to support cities

File: app/javascript/src/components/Profile/Organization/Edit/index.tsx [hunk 1]
```diff
@@ -325,6 +352,15 @@ const OrgEdit = () => {
     setStateList(stateList);
   }, [orgDetails.companyAddr.country]);
 
+  useEffect(() => {
+    setCurrentCityList(
+      City.getCitiesOfState(
+        orgDetails.companyAddr.country.code,
+        orgDetails.companyAddr.state.code ?? orgDetails.companyAddr.state.value
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Edit/index.tsx:359]: use optional chaining

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Review by prasanthchaduvula — APPROVED** [review]

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**